### ISSUE TITLE
ClassNotFoundException: com.fsck.k9.search.LocalSearch

### ISSUE THREAD

**Issue by philipwhiuk** [thunderbird/thunderbird-android#1240]
### Expected behaviour

Classes should exist
### Actual behaviour

```
03-31 12:40:23.101 E/Parcel  (  927): Class not found when unmarshalling: com.fsck.k9.search.LocalSearch^M
03-31 12:40:23.101 E/Parcel  (  927): java.lang.ClassNotFoundException: com.fsck.k9.search.LocalSearch^M
03-31 12:40:23.101 E/Parcel  (  927):   at java.lang.Class.classForName(Native Method)^M
03-31 12:40:23.101 E/Parcel  (  927):   at java.lang.Class.forName(Class.java:308)^M
03-31 12:40:23.101 E/Parcel  (  927):   at java.lang.Class.forName(Class.java:272)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.os.Parcel.readParcelableCreator(Parcel.java:2275)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.os.Parcel.readParcelable(Parcel.java:2239)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.os.Parcel.readValue(Parcel.java:2146)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.os.Parcel.readArrayMapInternal(Parcel.java:2479)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.os.BaseBundle.unparcel(BaseBundle.java:221)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.os.BaseBundle.getString(BaseBundle.java:918)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.content.Intent.getStringExtra(Intent.java:5378)^M
03-31 12:40:23.101 E/Parcel  (  927):   at com.android.server.am.ActivityStackSupervisor.startActivityLocked(ActivityStackSupervisor.java:1759)^M
03-31 12:40:23.101 E/Parcel  (  927):   at com.android.server.am.ActivityStackSupervisor.startActivityMayWait(ActivityStackSupervisor.java:1305)^M
03-31 12:40:23.101 E/Parcel  (  927):   at com.android.server.am.ActivityManagerService.startActivityAsUser(ActivityManagerService.java:4616)^M
03-31 12:40:23.101 E/Parcel  (  927):   at com.android.server.am.ActivityManagerService.startActivity(ActivityManagerService.java:4454)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:142)^M
03-31 12:40:23.101 E/Parcel  (  927):   at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:3021)^M
03-31 12:40:23.101 E/Parcel  (  927):   at android.os.Binder.execTransact(Binder.java:446)^M
03-31 12:40:23.101 E/Parcel  (  927): Caused by: java.lang.ClassNotFoundException: Didn't find class "com.fsck.k9.search.LocalSearch" on path: DexPathList[[directory "."],nativeLibraryDirectories=[/vendor/lib, /system/lib]]^M
03-31 12:40:23.101 E/Parcel  (  927):   at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)^M
03-31 12:40:23.101 E/Parcel  (  927):   at java.lang.ClassLoader.loadClass(ClassLoader.java:511)^M
03-31 12:40:23.101 E/Parcel  (  927):   at java.lang.ClassLoader.loadClass(ClassLoader.java:469)^M
03-31 12:40:23.101 E/Parcel  (  927):   ... 17 more^M
03-31 12:40:23.101 E/Parcel  (  927):   Suppressed: java.lang.ClassNotFoundException: com.fsck.k9.search.LocalSearch^M
03-31 12:40:23.101 E/Parcel  (  927):           at java.lang.Class.classForName(Native Method)^M
03-31 12:40:23.101 E/Parcel  (  927):           at java.lang.BootClassLoader.findClass(ClassLoader.java:781)^M
03-31 12:40:23.101 E/Parcel  (  927):           at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)^M
03-31 12:40:23.101 E/Parcel  (  927):           at java.lang.ClassLoader.loadClass(ClassLoader.java:504)^M
03-31 12:40:23.101 E/Parcel  (  927):           ... 18 more^M
03-31 12:40:23.101 E/Parcel  (  927):   Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available
```
### Steps to reproduce
1. TBD
### Environment

K-9 Mail version: TBD

Android version: TBD

Account type (IMAP, POP3, WebDAV/Exchange): N/A / TBD

**Comment by philipwhiuk** [comment]
@amennini Can you please tell us how you installed the app and what version it is?

**Comment by amennini** [comment]
I've installed from Google Play, version is 5.008. Phone is a Samsumg Galaxy S5, Android is 5.0 original, no root, completely standardo phone

**Comment by philipwhiuk** [comment]
I'm hoping this was a packaging error. I'm thinking it's going to be hard to reproduce or fix. Tempted to close - feel free to re-open if you reproduce this on a later version (e.g. 5.010+)

**Comment by philipwhiuk** [comment]
Well I saw this one again in a log for #1998 

And I'm willing to bet some money it's responsible for: #2019 

@cketti could this be a packaging error? Do we need to investigate the APK at this point?

**Comment by cketti** [comment]
This is probably a bug in Android.

**Comment by cketti** [comment]
http://stackoverflow.com/questions/28589509/android-e-parcel-class-not-found-when-unmarshalling-only-on-samsung-tab3

**Comment by cketti** [comment]
https://commonsware.com/blog/2016/07/22/be-careful-where-you-use-custom-parcelables.html :(

**Comment by Valodim** [comment]
Convert to byte array seems like the easiest solution :/

**Comment by philipwhiuk** [comment]
Should be fixed by #2169
